### ISSUE TITLE
Use a minimal runtime for binary executables

### ISSUE THREAD

**Issue by sv2dev** [oven-sh/bun#14546]
### What is the problem this feature would solve?

The resulting bundle size of the `bun build --compile` command is very large.
With Bun 1.1.30, for a hello world program it's already 57MB, targeting `bun-darwin-arm64`.
The program consists of a single file:
```ts
console.log("Hello, World!");
```
The command I used:
```
bun build --compile --minify ./index.ts --target=bun-darwin-arm64 --outfile app
```
This means that the bundled runtime itself has to be very large.

### What is the feature you are proposing to solve the problem?

I'm not sure what is included in the runtime for the single-file executables.
There are many features included in Bun that are great for the DX, but not needed at execution time in production, e.g. bundling, testing, package management, etc.
Also runtime APIs like `bun:sqlite`, and maybe also node APIs are not needed, unless they are explicitly used.

**Comment by KilianB** [comment]
Java Script Code is bundled with the binary which itself is pretty big and means that you do not need to have bun installed on the target system. I am strongly in favour of a feature to strip as many modules as possible, but I am not sure if the bun project itself is really the way to go or if this would need to be a dedicated fork.

I wish I could use typescript for embedded development but the compiled bundle just doesn't fit on the tiny devices.

**Comment by RedWilly** [comment]
Is this still be looked into??

**Comment by shubsharan** [comment]
Would really love to see some progress on this front. I'm building a CLI app I would like to publish to npm, only I want users to be able to use npx for the published package without me needing to upload a bunch of massive executables that all contain the entire runtime. I would like to use some of bun's runtime APIs because of the speed advantage vs node alternatives, but this is a major blocker for me.

**Comment by exoRift** [comment]
Yes, I would heavily be in favor of a compilation setting for aggressive runtime feature stripping. This is one of the large obstacles to using Bun to create applications
